### ISSUE TITLE
Context binding ambiguity?

### ISSUE THREAD

**Issue by guaiamum** [localForage/localForage#1029]
So there I was moving things around in an app using redux-saga, and faced [this issue](https://stackoverflow.com/questions/51319230/redux-saga-yield-call-throw-error-typeerror-cannot-read-property-ready-of-nul).

For what I could tell, the issue is that when calling `setItem` (or `getItem`), the context of self is not what we expect.
A quick/simple fix lies in [my reply](https://stackoverflow.com/a/68358403), but IMO we shouldn't have to do this. The context of the db should always be the same no matter how we call it, unless we forcefully bind it (as I did on the stack overflow reply).

Would love your comments on this, and perhaps a proposed solution that I could work on :)

<img width="737" alt="localforage context" src="https://user-images.githubusercontent.com/21321095/125424183-f839db3f-6b34-433c-a9b0-248e996017d4.png">

**Comment by tofumatt** [comment]
In the V2 rewrite this type of behaviour is removed/simplified, but I haven't had the chance to finish it yet. I don't have a good proposal to fix the V1 codebase in the meantime, because honestly it's quite old at this point and I find it a bit hard to follow for larger refactors 😅

**Comment by guaiamum** [comment]
Haha, no worries, glad to know we have a fix on the way! Closing this one then!
